### ISSUE TITLE
Invalid left operand

### ISSUE THREAD

**Issue by clementval** [omni-compiler/omni-compiler#208]
The following code gives an error at statement -> `array1(i,j) = values_ref%values2(1)`.

```fortran
MODULE mod1

IMPLICIT NONE

INTEGER, PARAMETER :: nb_max = 10
INTEGER :: i
TYPE typ1
  INTEGER :: nb
  REAL :: values1(nb_max)
  REAL :: values2(nb_max)
END TYPE typ1

TYPE (typ1), PARAMETER ::        & 
  values_ref = typ1(1,           &
    (/1.0, (0.0,i=2,nb_max)/),   &
    (/3.0, (1.E+10,i=2,nb_max)/))


CONTAINS
  SUBROUTINE sub1
    IMPLICIT NONE
    INTEGER :: i, j
    REAL :: array1(10, 10)
    DO j = 1, 10
      DO i = 1, 10
        array1(i,j) = values_ref%values2(1)
      END DO
    END DO
  END SUBROUTINE sub1
END MODULE mod1
```

```bash
"mod1.f90:mod1", line 26: invalid left operand of '%'
```

Looks like an `F95_STRUCT_CONSTRUCTOR` is returned and is should be just a member ref I guess.

**Comment by clementval** [comment]
If the `PARAMETER` attribute is removed, this example works.

**Comment by clementval** [comment]
Simpler example that reproduces the bug as well. 

```fortran 
MODULE mod1

IMPLICIT NONE

TYPE typ1
  INTEGER :: nb
END TYPE typ1

TYPE (typ1), PARAMETER :: values_ref = typ1(1)

CONTAINS
  SUBROUTINE sub1
    IMPLICIT NONE
    INTEGER :: array1(10)
    array1(:) = values_ref%nb
  END SUBROUTINE sub1
END MODULE mod1
```

**Comment by clementval** [comment]
PR #212 waiting for merge

**Comment by clementval** [comment]
Closed with #212
